src/RegistrationController.js

```diff
@@ -87,53 +137,80 @@ function (
             this.set('activationToken', resp.activationToken);
             delete resp.activationToken;
             return resp;
+          },
+          save: function() {
+            if(_.isFunction(preSubmit)) {
+              //async callback
+              preSubmit(this.attributes, function() {
```

Comment:
Let's try to come up with a better way to write this logic. As is, it's pretty tangled up:

- We have a `presubmit` function that they may or may not have defined, so we've two paths
  - Already here, we can simplify by following the pattern we use elsewhere where user input is normalized up-front in the settings.js file (i.e. transformUsername). We shouldn't have to check if they've defined something way down here in the registration controller.
- That calls `postFormSubmit`
- That calls the model save, which then calls `doFormSubmit` 
- That has some logic to once again parse the config input for `postSubmit` (like above, we should normalize this before it even hits this function in settings.js), which then calls `doPostSubmit`
- And then that does some stuff

And each of these places are handling errors, etc. Take a first pass, and if you want some help let's talk about it.

Reply:
Agreed. Will try to move the logic inside the callbacks into settings.